### PR TITLE
Added script to install kustomize from source

### PR DESCRIPTION
**Pull request by murphybytes**
# Description

The kustomize script to install binaries was broken which in turn broke our Makefile.  The PR adds a script to install kustomize from source if kustomize is not already installed.  It has a number of advantages over the old way of doing things.

1. Tagged source history is almost always immutable, the likelihood that a breaking change will be introduced from the kustomize project is low. 
1. This script will not silently overwrite a user's existing kustomize install like the old install would. 
1. The criticism that this approach will increase build times is unwarranted.  While it is true that the install will increase build time, it's the increase would be no more than importing an additional package into the project which we do all the time without a second thought.  On a development machine, it's a one time 4 second hit. 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Chore (documentation addition or typo, file relocation)

## Testing

- [ ] New tests were added with this PR that prove my fix is effective or that my feature works (describe below this bullet)
- [x] This change requires no testing (i.e. documentation update)

## Documentation

- [ ] All added public packages, funcs, and types have been documented with doc comments
- [ ] I have commented my code, particularly in hard-to-understand areas

## Final Checklist:

- [ ] I followed standard [GitHub flow](https://guides.github.com/introduction/flow/) guidelines
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings

## Additional Information (omit if empty)

Please include anything else relevant to this PR that may be useful to know.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -53,8 +53,7 @@ install-kubebuilder:
 
 .PHONY: install-kustomize
 install-kustomize:
-	curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | bash -s 3.8.6
-	mv kustomize ${KUSTOMIZE_INSTALL_DIR}/
+	@script/install-kustomize.sh
```
Comment (aleksej-paschenko) [inline Makefile:56]: if it does not take the destination directory, will [this](https://github.com/shipa-corp/ketch/blob/main/.github/workflows/release.yaml#L35) work?

Comment (murphybytes) [reply Makefile:56]: Nope, good catch.  I'll fix it.

**Comment by vivek** [comment]
This issue with kustomize is no longer present, ok to close this PR @murphybytes?

**Comment by vivek** [comment]
Closing this as kustomize issue is not more present, we can revisit if it becomes issue.

**Pull request closed without merging**